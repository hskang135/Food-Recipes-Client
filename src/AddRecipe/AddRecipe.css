#foodname,
#ingredients,
#description {
  border: 0.1rem solid black;
  padding: 0.5rem;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-shadow: 2px 2px 5px #434343;
  font-size: 1.1rem;
}

#ingredients,
#description {
  height: 9rem;
  width: 14rem;
}

.AddRecipe_form {
  border: 0.1rem solid black;
  max-width: 30rem;
  padding: 1.5rem;
  text-align: center;
  margin: 1.6rem;
  font-size: 1.3rem;
  font-family: 'Nunito', sans-serif;
}

@media screen and (min-width: 768px) {
  .AddRecipe_form {
    margin: auto;
    max-width: 37rem;
    margin-bottom: 6rem;
  }

  #foodname,
  #ingredients,
  #description {
    padding: 0.5rem;
    margin-top: 1.5rem;
    margin-bottom: 1.5rem;
    box-shadow: 2px 2px 5px #434343;
    font-size: 1.5rem;
  }

  #ingredients,
  #description {
    height: 10rem;
    width: 30rem;
  }
}

@media screen and (min-width: 1024px) {
  h3 {
    font-size: 2.7rem;
  }

  label {
    font-size: 1.8rem;
  }

  .AddRecipe_form {
    margin: auto;
    max-width: 50rem;
    margin-bottom: 6rem;
  }

  #foodname,
  #ingredients,
  #description {
    padding: 0.5rem;
    margin-top: 1.5rem;
    margin-bottom: 1.5rem;
    box-shadow: 2px 2px 5px #434343;
    font-size: 1.5rem;
  }

  #foodname {
    width: 23rem;
  }

  #ingredients,
  #description {
    height: 10rem;
    width: 35rem;
  }
}



