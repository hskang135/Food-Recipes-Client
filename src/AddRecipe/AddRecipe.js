import React from 'react';

export default function AddRecipe(props) {

  return(
    <form className="create-recipe">
        <h2>Create New Recipe</h2>
        <div className="form-section">
          <label htmlFor="recipes-title">Name of Recipe Title: </label>
          <input type="text" name="recipes-title" placeholder="Add food recipe name" required />
        </div>

        <div className="form-section">
          <label htmlFor="ingredients-summary">Ingredients: </label>
          <textarea name="ingredients-summary" rows="5" required></textarea>
        </div>
          
        <div className="form-section">
          <label htmlFor="recipes-summary">Recipe Summary: </label>
          <textarea name="recipes-summary" rows="5" required></textarea>
        </div>
          
        <button type="submit">Submit New Recipe</button>
      </form>
  )

}
