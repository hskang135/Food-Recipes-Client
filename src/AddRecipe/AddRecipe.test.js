import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddRecipe from './AddRecipe'

describe(`AddReceipe component`, () => {
  const stubAddRecipe = [
    {
      "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "foodname": "Apple water",
      "ingredients": "Test ingredients",
      "description": "Add all ingredients"
    },
    {
      "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
      "foodname": "Coffee bread",
      "ingredients": "Test ingredients",
      "description": "Combine all ingredients and put in the oven"
    }
  ]

  it('renders the complete form', () => {
    const context = { rescipes: stubAddRecipe }
    const wrapper = shallow(<AddRecipe />, context)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
})


