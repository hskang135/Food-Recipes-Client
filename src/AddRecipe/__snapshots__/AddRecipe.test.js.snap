// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReceipe component renders the complete form 1`] = `
<section
  className="AddRecipe"
>
  <h3>
     Create New Recipe 
  </h3>
  <form
    className="AddRecipe_form"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="Name"
      >
         Food Recipe Name: 
         
         
        <Required />
      </label>
       
      <br />
      <input
        id="foodname"
        name="recipeName"
        placeholder="Food Recipe Name"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="Ingredients"
      >
        Ingredients 
        <br />
        (Please write seperate by comma): 
         
         
        <Required />
      </label>
       
      <br />
      <textarea
        cols="45"
        id="ingredients"
        name="ingredients"
        required={true}
        rows="6"
      />
    </div>
    <div>
      <label
        htmlFor="Description"
      >
         Description: 
         
         
        <Required />
      </label>
       
      <br />
      <textarea
        cols="45"
        id="description"
        name="description"
        required={true}
        rows="6"
      />
    </div>
    <div>
      <button
        type="submit"
      >
         Submit New Recipe 
      </button>
    </div>
  </form>
</section>
`;
